### ISSUE TITLE
Rename RealmConfiguration.Builder.name() to filename()

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#2699]
We had a couple of cases where people have been confused by the `RealmConfiguration.Builder.name()` parameter. Especially they expected the filename to end with `.realm`.

We don't have any techical requirements for enforcing that realm files end with `.realm`, and the only advantage would be that e.g. any stand-alone browser could more easily identify the file as a Realm file.

IMO we could do a number of things to make this slightly easier:

1) Automatically append `.realm` to any name if it doesn't have it already. 

2) Always append `.realm`. This means that existing apps might start seeing `foo.realm.realm`. Would also make upgrading a bit tricky as it would be easy to forget. 

3) Keep current behaviour, but change the method to `filename(String)` instead of `name(String)` as that would make the intent more clear.

I would probably lean towards 3) as that makes the intent clear without breaking any existing files. I could be convinced that 1) also might be okay, since having well defined file extensions is always a plus if people are managing the file themselves somehow.

What do you think @realm/java ?

**Comment by kneth** [comment]
I agree: option 3 is best. Maybe we need to update the documentation too?

**Comment by cmelchior** [comment]
Yeah, we should clarify in the javadoc as well

**Comment by zaki50** [comment]
I prefer 1), but 1) may cause some problems.

So, I vote 3)

**Comment by stk1m1** [comment]
As a conservative, my vote goes to :three:

**Comment by dalinaum** [comment]
I prefer option 3.

**Comment by diegomontoya** [comment]
I like 3.
